src/Nethermind/Nethermind.BeaconNode.Peering/MothraPeeringWorker.cs

```diff
@@ -212,6 +236,53 @@ await JsonSerializer.SerializeAsync(fileStream, signedBeaconBlock, _jsonSerializ
             }
         }
 
+        private async void HandlePeerDiscoveredAsync(string peerId)
+        {
+            try
+            {
+                if (_peerManager.AddPeer(peerId))
+                {
+                    // Delay to ensure connection is established
+                    await Task.Delay(TimeSpan.FromMilliseconds(500)).ConfigureAwait(false);
+
+                    await _synchronizationManager.OnPeerDialOutConnected(peerId).ConfigureAwait(false);
+                }
+            }
+            catch (Exception ex)
+            {
+                if (_logger.IsError())
+                    Log.HandlePeerDiscoveredError(_logger, peerId, ex.Message, ex);
+            }
+        }
+
+        private async void HandleRpcStatusAsync(RpcMessage<PeeringStatus> statusRpcMessage)
+        {
+            try
+            {
+                PeerDetails peerDetails = _peerManager.UpdatePeerStatus(statusRpcMessage.PeerId, statusRpcMessage.Content);
+                // Mothra seems to be raising all incoming RPC (sent as request and as response)
+                // with requestResponseFlag 0, so check here if already have the status so we don't go into infinite loop
+                // So user peerdetails instead of the status message
+                //if (statusRpcMessage.Direction == RpcDirection.Request)
+                if (peerDetails.DialDirection == DialDirection.DialOut)
+                {
```

Comment:
one big issue we had in Eth1 was when two peers were connecting to each other at the same time (happening very very often) - they should pick properly which connection to keep and which to drop - I think that we can almost reuse big part of the peer management code between Eth1 and Eth2 after some refactoring - maybe you can review the current peer manager and see if it is possible to improve / merge both with proper abstractions?

Reply:
Yes, good idea.

Reply:
Probably still too early to directly use objects -- Eth2 is (currently) very simple compared to Eth1 (multiple sync protocols, etc), and pretty much different from the ground up (on purpose), e.g. different P2P, different encryption, etc.

Reply:
I've reused concepts from Eth1, so we can add the complexity later. Hopefully it is looking better now.